test/testdrive/tables.td

```diff
@@ -197,6 +197,12 @@ a       b
 ! INSERT INTO s VALUES (1 + NULL);
 contains:null value in column "a" violates not-null constraint
 
+! INSERT INTO s VALUES (case when now() = now() then NULL else 2 end);
+contains:null value in column "a" violates not-null constraint
+
+! INSERT INTO s SELECT (case when now() = now() then NULL else 2 end);
```

Comment:
Do we support `INSERT .. SELECT * FROM another_table?` If so a test about that would appease me. I don't have a mental model of what things might be const-ified, but a read of another table definitely can't.

Reply:
We do, I've added a test case with that.